src/operator/rnn_impl.h

```diff
@@ -994,7 +998,6 @@ void GruForwardTraining(DType* ws,
   DType* bx_l = bx;
   DType* bh_l = bh;
   DType* y_tmp = x_ptr;
-  unsigned int seed_ = 17 + rand() % 4096;  // NOLINT(runtime/threadsafe_fn)
```

Comment:
I think this line is copied from https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/dropout-inl.h#L85. So do you think we also need make change in dropout?

Reply:
Well, the problem is not about the seed itself, but with usage of `rand_r` and the fact that it's multithreaded. As far as I can tell in dropout-inl.h#L85 the seed is set outside of OpenMP threads and no random number gets generated with it afterwards, so it should be fine.